airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java

```diff
@@ -632,6 +632,8 @@ public BufferedImage createImage(int size) {
                     throw new NullPointerException();
                 }
                 return result;
+            } catch (NullPointerException x) {
+                LOG.warn("Failed to process cover art for " + mediaFile + ": " + x);
```

Comment:
I think this should just be changed such that the NPE isn't thrown in the first place. Maybe do:

```
if result != null:
  return result
else:
  LOG.warn...

Reply:
The NPE is meant to catch (and warn) about the problem while still ending up falling back to the call to `createAutoCover()`